src/frontend/screens/ObservationEdit/SaveButton.tsx

```diff
@@ -69,20 +71,43 @@ export const SaveButton = ({
   const navigation = useNavigationFromRoot();
   const createObservationMutation = useCreateObservation();
   const editObservationMutation = useEditObservation();
+  const createAttachmentsMutation = useCreateAttachmentsMutation();
 
   function createObservation() {
     if (!value) throw new Error('no observation saved in persisted state ');
-    createObservationMutation.mutate(
-      {value},
-      {
-        onError: () => {
-          if (openErrorModal) openErrorModal();
-        },
-        onSuccess: () => {
-          navigation.navigate('Home', {screen: 'Map'});
-        },
+
+    const savablePhotos = photos.filter(isSavablePhoto);
+
+    createAttachmentsMutation.mutate(savablePhotos, {
```

Comment:
Do we need to do a check for instances where there are no photos? And then if there are no photos, skip creating the attachment, or will the empty array handle that?

Reply:
the empty array should handle that but if we're looking to optimize a bit, we can do the check and completely avoid calling the mutation in the case of no savablePhotos 👍

Reply:
addressed via https://github.com/digidem/CoMapeo-mobile/pull/187/commits/98f0255207d762169a288d57d093d91509d705b1